src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -297,6 +350,44 @@ private IContainer calculateGlobalPKAnalysisFor(Simulation simulation, Compound
          return container;
       }
 
+      private IParameter cmaxRatioCalculator(int individualId, CompoundPK compoundPKFromContext, float? cMax)
+      {
+         var parameter = contextParameterFrom(CoreConstants.PKAnalysis.C_maxRatio, compoundPKFromContext.QuantityFor(individualId, CoreConstants.PKAnalysis.C_maxRatio),
+            cMax, compoundPKFromContext.CMaxDDIFor(individualId));
+
+         compoundPKFromContext.AddDDICMax(individualId, cMax);
+         return parameter;
+      }
+
+      private IParameter aucRatioCalculator(int individualId, CompoundPK compoundPKFromContext, float? aucInf)
+      {
+         var parameter = contextParameterFrom(CoreConstants.PKAnalysis.AUCRatio, compoundPKFromContext.QuantityFor(individualId, CoreConstants.PKAnalysis.AUCRatio),
+            aucInf, compoundPKFromContext.DDIAucInfFor(individualId));
+
+         compoundPKFromContext.AddDDIAucInf(individualId, aucInf);
+
+         return parameter;
+      }
+
+      private IParameter bioAvailabilityCalculator(int individualId, CompoundPK compoundPKFromContext, double? aucInf)
```

Comment:
wondering about double here and why not float

Reply:
all those calculators are looking very much the same. Can we extract one one method?

Reply:
import static in this file. It will be so much easier to read

Reply:
> all those calculators are looking very much the same. Can we extract one one method?

That's just removing these functions. They all call the same method to create the parameter